Nouveau portail

mamakh opened the pull request:
## Description

Design : PR #1
Mise en place du nouveau portail.
Le design a complètement été revu avec une nouvelle charte graphique, de nouvelles couleurs, un nouveau header, un nouveau footer.
Les couleurs et les logos sont toujours liées au  moteur de thèmes.
Le moteur d'applications a été revu également.
L'affichage des catégories a été revu.

## Type de changement

-  Code et/ou Configuration : Nouvelle fonctionnalité
-  Code et/ou Configuration : Refactorisation
-  Nouvelle fonctionnalité (modification cassante rajoutant des fonctionnalités)
-  Ansiblerie

## Tests

Tests d'affichage sur le portail.


## Checklist

 [Xelians] Mon code suit le style de code de ce projet.
 [x] J'ai commenté mon code, en particulier dans les classes et les méthodes difficile à comprendre.
 [x] J'ai rajouté les tests unitaires vérifiant mes fonctionnalités.
 [x] J'ai rajouté les tests de non régression vérifiant mes fonctionnalités.
 [x] Les tests unitaires nouveaux et existants passent avec succès localement.
 [x] Toutes les dépendances ont été mergées en priorité

## Contributeur

Xelians

Commit:
Add Categories Configuration
Add deployment config + unit test

Commit:
Fix jinja templating for categories + cleanup

fadil-zemmari-xelians committed:
[US TRTL-144] : Add footer component

[[US TRTL-144] : add customer info

[US TRTL-144] : Code review

[TECH] Add new portal development page

fadil-zemmari-xelians committed:
[TECH] Add new portal development page

fadil-zemmari-xelians committed:
[US TRTL-146 TRTL-148] Add new portal application list design

youcefxelians committed:
[US TRTL-147] : Add scrollTop component

[US TRTL-147] : code review

fadil-zemmari-xelians committed:
[US TRTL-146 TRTL-148] Add new portal application list design


[US TRTL-145] Add welcome message in new portal application


[US TRTL-149] Add filter by categorie

fadil-zemmari-xelians committed:
[US TRTL-145] Add welcome message in new portal application


[US TRTL-145] Rename welcome message to welcome description

youcefxelians committed:
[US TRTL-167] : add new footer design

[US TRTL-167] : code review

fadil-zemmari-xelians committed:
[US TRTL-143] add new portal header


[US TRTL-143] Add header design


[US TRTL-143] Fix header after code review


[US TRTL-143] Add a module for tenant selection component

fadil-zemmari-xelians committed:
[TECH] Assemble and fix new portal component


[TECH] Fix deploy


[TECH] Fix new portal responsive display


[US TRTL-163] Add new coloured applications icon


[TECH] Fix VVitamUI vars


[TECH] Fix VitamUI vars
[TECH] Fix new portal responsive display

fadil-zemmari-xelians committed:
[US TRTL-189] Add common body container

Commit:
[US TRTL-150] Add Application menu launcher

[US TRTL-150] without overlay
[US TRTl-150] changes after new spec
[US TRTL-150] review css
[TECH] temp rebase portal
[US TRTL-150] move in fronend-common & fix highlightPipe & fix select-tenant
[US TRTL-150] fix tslint
[US TRTL-150] fix modules
[US TRTL-150] integrate menu button

fadil-zemmari-xelians committed:
[TECH] Add application sorting on new portal design

fadil-zemmari-xelians committed:
[TECH] Sort categories in applications overlay


[TECH] Fix select tenant component NP error


[TECH] Fix scroll top component appear


[TECH] Move new portal to portal component

fadil-zemmari-xelians committed:
[US TRTL-200] change select tenant

[FIX TRTL-217] Fix wording

fadil-zemmari-xelians committed:
[US TRTL-175] Mutualise group by application and routing


[US TRTL-175] Refactoring group by application and routing


[US TRTL-175] Review improvement

fadil-zemmari-xelians committed:
[US TRTL-199] Performance improvement on portal

[US TRTL-199] Lighthouse SEO improvement
[US TRTL-199] Lighthouse Performance improvement
[US TRTL-199] Performance improvement
[US TRTL-199] Fix rebase conflict
[TECH] Fix css load defer delay

fadil-zemmari-xelians committed:
[FIX TRTL-237] Fix application service for identity apps access


[US TRTL-235] Search apps on change in menu component


[US TRTL-235] Code review

Commit:
[US TRTL-280] Test new application SVG icon


[US TRTL-280] Add SVG icons MANUAL_INGEST_APP & ARCHIVE_APP


[US TRTL-334] Fix apps icon dynamic color

fadil-zemmari-xelians committed:
[US TRTL-431] Remove moment-mini dependency

Commit:
[TECH] Change categories format and fix display

Commit:
[TECH] Hide category with no app

and init user category name as "Utilisateur"

Commit:
[TECH] Change categories format and fix display

Commit:
Merge branch 'feature/turtle/portal' into dlab-develop

Mocteezuma commented on ui/ui-frontend-common/src/app/modules/components/footer/footer.component.spec.ts:
```diff
@@ -0,0 +1,22 @@
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
```
Est-ce normal d'avoir un TU sans test ?
Je comprend que le footer n'ait pas grand chose a tester mais dans ce cas le création du fichier est-elle pertinante ?
Si oui, ne peut on pas au moins tester que le composant se crée bien ?

mamakh replied:
cf https://github.com/ProgrammeVitam/vitam-ui/pull/66#discussion_r539351980

Mocteezuma commented on ui/ui-frontend-common/src/app/modules/components/header/menu/menu.component.html:
```diff
@@ -0,0 +1,46 @@
+        ></vitamui-common-search-bar>
+        <span class="pr-2" i18n="menuComponent@@myApplications">Mes applications</span>
+        <button (click)="onClose()" mat-icon-button class="apps-button">
+            <i class="vitamui-icon vitamui-icon-apps-colored"><span class="path1"></span><span class="path2"></span><span class="path3"></span><span class="path4"></span><span class="path5"></span><span class="path6"></span><span class="path7"></span><span class="path8"></span><span class="path9"></span></i>
```
C'est quoi ces classes de path ? Que porte t-elles ?

A quoi servent ells ? Et pourquoi en initialiser autant (en particulier pour MANUAL_INGEST_APP où on en initialiser 188 ?)
Si c'est pour lister des éléments pourquoi ne pas utiliser `*ngFor in` ?

delphine-coursier replied:
Les classes de path sont générées automatiquement par la librairie Icomoon lors de l'ajout d'une icone complexe, comme c'est le cas pour les nouvelles icones des applications ^^ L'icône globale est composée de multiples entitées traduites chacune comme des "path", comme si c'était une grosse icone composée de sous icônes. Les paths sont déclarés dans le fichier des classes icomoon vitamui-icons.css

Mocteezuma commented on ui/ui-frontend/projects/portal/src/app/components/application-list/application-list.component.html:
```diff
@@ -0,0 +1,35 @@
+</div>
+
+<ng-template #icons let-application="application">
+  <div *ngIf="application.identifier === 'ARCHIVE_APP'">
+    <i class="vitamui-icon vitamui-icon-app_archive"><span class="path1"></span><span class="path2"></span><span class="path3"></span><span class="path4"></span><span class="path5"></span><span class="path6"></span><span class="path7"></span><span class="path8"></span><span class="path9"></span><span class="path10"></span><span class="path11"></span><span class="path12"></span><span class="path13 ...
+  </div>
+
+  <div *ngIf="application.identifier === 'MANUAL_INGEST_APP'">
+    <i class="vitamui-icon vitamui-icon-app_sip_seda"><span class="path1"></span><span class="path2"></span><span class="path3"></span><span class="path4"></span><span class="path5"></span><span class="path6"></span><span class="path7"></span><span class="path8"></span><span class="path9"></span><span class="path10"></span><span class="path11"></span><span class="path12"></span><span class="path1 ...
```
Pareil qu'au dessus, d'où viennent ces classes "pathX" et a quoi servent elles ?

mamakh replied:
cf https://github.com/ProgrammeVitam/vitam-ui/pull/66#discussion_r539332209

Mocteezuma commented on ui/ui-frontend/projects/portal/src/app/components/application-list/application-list.component.spec.ts:
```diff
@@ -0,0 +1,36 @@
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
```
Fichier (de test) tout vide ?

mamakh replied:
cf https://github.com/ProgrammeVitam/vitam-ui/pull/66#discussion_r539351980

Mocteezuma commented on ui/ui-frontend/projects/portal/src/app/components/welcome-message/welcome-message.component.spec.ts:
```diff
@@ -0,0 +1,36 @@
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
```
Fichier (de test) vide ?

delphine-coursier replied:
Notre politique de test front est de supprimer les fichiers de test vides, mais certains ont vu dans Vitam-ui qu'il y avait des fichiers de test vides avec juste la licence et ont tout simplement fait pareil ^^ mais on peut très bien refaire une passe et les supprimer !

mamakh replied:
on peut proposer une PR à la suite pour supprimer les fichiers de tests vides sur l'ensemble du projet

Mocteezuma replied:
Pourquoi pas, ok

Mocteezuma reviewed (commented):
Une première review pour avancer sur le sujet en attendant la présentation de jeudi.

Commit:
[TECH] Remove tar.gz assembly

pyXelians committed:
Merge branch 'feature/design/0' into feature/design/1

pyXelians committed:
[TECH] Fix version vitamui in ingest module

mamakh committed:
Mise en place du changement de langue avec ngx-translate (#68)

* [US TRTL-157] component SelectLanguage & VitamTranslateService

[US TRTL-157] MultiLoadHttpLoader & translate templates

[US TRTL-157]add gulpfile to migrate from i18n to ngx-translate

[US TRTL-157] save/update language & sessionStorage

[US TRTL-157] sessionStorage & add ROLE_UPDATE_ME_USERS for portal & delete gulpfile.js

[US TRTL-157] clean i18n for portal in conf files

[US TRTL-157] review & fix ngx-translate version

* [US TRTL-427] add httpLoaderChildFactory to use local assets/i18n

* [TRTL-484] Fix ngx translate with lazy modules

* [US TRTL-473] Hide selection language button

* [TECH] Fix ngx translate unit tests

* [TECH] Update categories for deployment & local configurations

* [TECH] PR review fix for moment-mini which was unused

* [Tech] Fix package-lock

* Mise en place des analytics utilisateur et de la catégorie 'Dernières applications utilisées' (#69)

* [US TRTL-157] component SelectLanguage & VitamTranslateService

[US TRTL-157] MultiLoadHttpLoader & translate templates

[US TRTL-157]add gulpfile to migrate from i18n to ngx-translate

[US TRTL-157] save/update language & sessionStorage

[US TRTL-157] sessionStorage & add ROLE_UPDATE_ME_USERS for portal & delete gulpfile.js

[US TRTL-157] clean i18n for portal in conf files

[US TRTL-157] review & fix ngx-translate version

* [US TRTL-427] add httpLoaderChildFactory to use local assets/i18n

* [TRTL-484] Fix ngx translate with lazy modules

* [US TRTL-473] Hide selection language button

* [TECH] Fix ngx translate unit tests

* [TECH] Update categories for deployment & local configurations

* [TECH] Remove tar.gz assembly

* [TECH] PR review fix for moment-mini which was unused

* [US TRTL-170] : save last used applications

[US TRTL-170] : code review - rename applicationAnalytics class

[US TRTL-170] : tag application us last used only at the first visit

[US TRTL-170] : code review

* [US TRTL-172] : Get last used app back-end

[US TRTL-172] : Get last used app front-end
[US TRTL-172] : Fix import

[US TRTL-172] : Get last used apps update
[US TRTL-172]: Fix after conflict resolve
[US TRTL-172]: Fix last used app analytics loading

[US TRTL-172]: Fix duplicate when searching an application

[US TRTL-172]: Code review improvement

* [TECH] : Delete logger debug in test

* [TECH] Fix review - Create applicationIDConstant + userBuilder to test

* [Tech] Fix package-lock

* [TECH] FIX merge

* Choix du tenant par défaut & Sélection du tenant ou de l'organisation pour les apps multi-tenants et multi-organisations (#71)

* [US TRTL-172] : Get last used app back-end

[US TRTL-172] : Get last used app front-end
[US TRTL-172] : Fix import

[US TRTL-172] : Get last used apps update
[US TRTL-172]: Fix after conflict resolve
[US TRTL-172]: Fix last used app analytics loading

[US TRTL-172]: Fix duplicate when searching an application

[US TRTL-172]: Code review improvement

* [US TRTL-170] : save last used applications

[US TRTL-170] : code review - rename applicationAnalytics class

[US TRTL-170] : tag application us last used only at the first visit

[US TRTL-170] : code review

* [US TRTL-171] Persist last tenant identifier analytics

* [US TRTL-216] Add select tenant dialog on first user auth

[US TRTL-216] Fix last tenant identifier synchronization

[US TRTL-216] Improvement & add tenant by app filter

[US TRTL-216] Code improvement
[US TRTL-216] Add hierarchy app new portal components

[US TRTL-216] Fix select tenant component css
[US TRTL-169] Add select tenant inside app engine
[TECH] Add lastly used application translation

[US TRTL-169] Fix selected tenant display in app engine

[US TRTL-169] Code & architecture improvement

[US TRTL-169] Add translation & code review

[US-TRTL-216] Revert header features from identity

* [US TRTL-408] Add unknown app icon


[FIX TRTL-423] Fix tenant selection


[US TRTL-425] Add customer selection in new header


[TRTL-425] Code review

* [TECH] Apply Translate modifications in selecttenant & selectcustomer

* [TECH] Remove dev files for VITAM (these files are in .gitignore)

* [TECH] Fix TU

* [TECH] Delete ngOnDestroy not used

Co-authored-by: Fadil Zemmari <fadil.zemmari@xelians.fr>
Co-authored-by: Delphine <delphine.coursier@teamdlab.com>
Co-authored-by: Makhtar DIAGNE <makhtar.diagne@teamdlab.com>
Co-authored-by: pybelecalo <pierre-yves.belecalo@xelians.fr>

Co-authored-by: Delphine <delphine.coursier@teamdlab.com>
Co-authored-by: Mickael <mickael.varlet@xelians.fr>
Co-authored-by: Fadil <Fadil.zemmari@xelians.fr>
Co-authored-by: Makhtar DIAGNE <makhtar.diagne@teamdlab.com>
Co-authored-by: pybelecalo <pierre-yves.belecalo@xelians.fr>

Co-authored-by: Delphine <delphine.coursier@teamdlab.com>
Co-authored-by: Mickael <mickael.varlet@xelians.fr>
Co-authored-by: Fadil <Fadil.zemmari@xelians.fr>
Co-authored-by: Makhtar DIAGNE <makhtar.diagne@teamdlab.com>
Co-authored-by: pybelecalo <pierre-yves.belecalo@xelians.fr>
